packages/react/src/firestore/useGetCountFromServerQuery.ts

```diff
@@ -0,0 +1,24 @@
+import { useQuery, type UseQueryOptions } from "@tanstack/react-query";
+import {
+  getCountFromServer,
+  type FirestoreError,
+  type Query,
+} from "firebase/firestore";
+
+type FirestoreUseQueryOptions<TData = unknown, TError = Error> = Omit<
+  UseQueryOptions<TData, TError>,
+  "queryFn"
+>;
+
+export function useGetCountFromServerQuery(
+  query: Query,
+  options: FirestoreUseQueryOptions<number, FirestoreError>
+) {
+  return useQuery<number, FirestoreError>({
+    ...options,
+    queryFn: async () => {
+      const snapshot = await getCountFromServer(query);
+      return snapshot.data().count;
```

Comment:
Should we be returning the snapshot here or the count?

Reply:
should be returning the snapshot. 

ive refactored to return an [AggregateQuerySnapshot](https://firebase.google.com/docs/reference/js/firestore_.aggregatequerysnapshot.md#aggregatequerysnapshot_class) which is described as the return type of the [getCountFromServer](https://firebase.google.com/docs/reference/js/firestore_.md#getcountfromserver_4e56953)